test/extensions/filters/http/wasm/wasm_filter_test.cc

```diff
@@ -691,7 +693,26 @@ TEST_P(WasmHttpFilterTest, AccessLogClientDisconnected) {
 
   Http::TestRequestHeaderMapImpl request_headers{{":path", "/"}};
   EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter().decodeHeaders(request_headers, false));
+
+  StreamInfo::MockStreamInfo log_stream_info;
+  EXPECT_CALL(log_stream_info, requestComplete())
+      .WillRepeatedly(testing::Return(std::chrono::milliseconds(30)));
+  filter().log(&request_headers, nullptr, nullptr, log_stream_info);
+  filter().onDestroy();
+}
+
+TEST_P(WasmHttpFilterTest, AccessLogDisabledForIncompleteStream) {
+  setupTest("", "headers");
+  setupFilter();
+  EXPECT_CALL(filter(),
+              log_(spdlog::level::debug, Eq(absl::string_view("onRequestHeaders 2 headers"))));
+  EXPECT_CALL(filter(), log_(spdlog::level::info, Eq(absl::string_view("header path /"))));
+  EXPECT_CALL(filter(), log_(spdlog::level::warn, Eq(absl::string_view("onDone 2"))));
+  Http::TestRequestHeaderMapImpl request_headers{{":path", "/"}};
+  EXPECT_EQ(Http::FilterHeadersStatus::Continue, filter().decodeHeaders(request_headers, false));
+
   StreamInfo::MockStreamInfo log_stream_info;
+  EXPECT_CALL(log_stream_info, requestComplete()).WillRepeatedly(testing::Return(absl::nullopt));
   filter().log(&request_headers, nullptr, nullptr, log_stream_info);
```

Comment:
Not sure this actually checks that Wasm log() is not called here? Is this the same check as for regular access log?